controllers/argocd/secret_test.go

```diff
@@ -595,6 +596,41 @@ func TestGenerateSortedManagedNamespaceListForArgoCDCR(t *testing.T) {
 	res, err = generateSortedManagedNamespaceListForArgoCDCR(a, r.Client)
 	assert.NoError(t, err)
 	assert.Equal(t, res, []string{managedByNamespace.Name, a.Namespace})
+
+	// If Namespace management is enabled, fetch from NamespaceManagement CRs
+	// and Ensure that results returned by this function are sorted by namespace name
+	os.Setenv(common.EnableManagedNamespace, "true")
+	defer os.Unsetenv(common.EnableManagedNamespace)
+
+	nsMgmt1 := argoproj.NamespaceManagement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "namespace-mgmt-1",
+			Namespace: "aaaa-second-when-sorted-nm",
+		},
+		Spec: argoproj.NamespaceManagementSpec{
+			ManagedBy: a.Namespace,
+		},
+	}
+
+	nsMgmt2 := argoproj.NamespaceManagement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "namespace-mgmt-2",
+			Namespace: "bbbb-third-when-sorted-nm",
+		},
+		Spec: argoproj.NamespaceManagementSpec{
+			ManagedBy: a.Namespace,
+		},
+	}
+
+	err = cl.Create(context.Background(), &nsMgmt1)
+	assert.NoError(t, err)
+	err = cl.Create(context.Background(), &nsMgmt2)
+	assert.NoError(t, err)
+
+	res, err = generateSortedManagedNamespaceListForArgoCDCR(a, r.Client)
+	assert.NoError(t, err)
+	assert.Equal(t, res, []string{managedByNamespace.Name, nsMgmt1.Namespace, a.Namespace, nsMgmt2.Namespace})
+
```

Comment:
Can you also verify that if there exists a NamespaceManagement that has `.spec.managedBy` that points to a different namespace, that the namespace of the NamespaceManagement does NOT appear in the list?

It would also be good to verify that namespace management namespaces don't appear in the list when the feature is disabled.